### PR TITLE
adds CloudEvent helloworld to examples, README, and tests

### PR DESCRIPTION
**Pull request by bshaffer**

### CODE REVIEW

**Review by grant — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -162,6 +162,64 @@ in the [Cloud Console][cloud-run-console].
 [cloud-run-regions]: https://cloud.google.com/run/docs/locations
 [cloud-run-console]: https://console.cloud.google.com/run
 
+## Use CloudEvents
+
+The Functions Framework can unmarshall incoming [CloudEvents][cloud-events]
+payloads to a `cloudevent` object. This will be passed as arguments to your
+function when it receives a request. Note that your function must use the
+cloudevent-style function signature:
```
Comment (ace-n) [inline README.md:170]: Nit: `cloudevent`?

File: examples/hello/composer.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
     "require": {
-        "google/cloud-functions-framework": "^0.3"
+        "google/cloud-functions-framework": "dev-master as 0.6.0"
```
Comment (ace-n) [inline examples/hello/composer.json:3]: Is this downloading the latest code available directly from GitHub? (i.e. disabling version pinning)

(This is a beta version, so I'm willing to allow it for now - but we **definitely** shouldn't do this once we release publicly.)

File: examples/hello/index.php [hunk 3]
```diff
@@ -15,10 +15,30 @@
  * limitations under the License.
  */
 
+/**
+ * To use this, set the following environment variables:
```
Comment (ace-n) [inline examples/hello/index.php:19]: Nit: can these also be specified as command-line args?

File: tests/exampleTest.php [hunk 4]
```diff
@@ -45,33 +45,77 @@ public static function setUpBeforeClass(): void
...
+            $outputAsString
+        );
+
+        passthru('docker rm -f ' . self::$containerId);
+        self::$containerId = null;
     }
 
     public static function tearDownAfterClass(): void
     {
+        // If a test failed before it could delete its container
```
Comment (ace-n) [inline tests/exampleTest.php:113]: Nit: would it be better to have a single "clean up test resources" comment?

(Or maybe even delete these comments outright, if you're of the "code should be self-documenting" philosophy.)

**Review by ace-n — APPROVED** [review]
LGTM once changes are addressed, **and** we have a future plan to re-enable version pinning (assuming that this PR disables it.)

File: examples/hello/composer.json (hunk 2 above)
Comment (bshaffer) [reply examples/hello/composer.json:3]: it's only for testing. I will change it back once a version is tagged

File: examples/hello/index.php (hunk 3 above)
Comment (bshaffer) [reply examples/hello/index.php:19]: there is no CLI for php at the moment

**Pull request merged**